sevntu-checks/pom.xml

```diff
@@ -37,6 +37,7 @@
       <groupId>com.puppycrawl.tools</groupId>
       <artifactId>checkstyle</artifactId>
       <version>${checkstyle.eclipse-cs.version}</version>
+      <scope>provided</scope>
```

Comment:
What does this mean to maven-checkstyle-plugin that included sevntu and never overrode the plugin's version of checkstyle?
This means that it will default to `maven-checkstyle-plugin`'s default checkstyle version where as before sevntu overrode the checkstyle version itself. Right?

> eclipsecs-sevntu-plugin doesn't include checkstyle jar
> sevntu-checkstyle-maven-plugin doesn't include checkstyle jar
> sevntu-checkstyle-idea-extension doesn't include checkstyle jar

If these don't include it, where does it get it's version of checkstyle from? When I looked at the POM's in eclipse, it listed checkstyle as a dependency from sevntu, but I don't see it in the created jars.
Is it like sonar and expects main checkstyle plugin to also be installed already?

> sevntu-checkstyle-maven-plugin

@romani How come we never used this plugin in main repo?

Reply:
https://github.com/checkstyle/checkstyle/blob/master/pom.xml#L554

Reply:
@rnveach ,
> What does this mean to maven-checkstyle-plugin that included sevntu and never overrode the plugin's version of checkstyle?

It means that default checkstyle version will be used from `maven-checkstyle-plugin`.

> This means that it will default to maven-checkstyle-plugin's default checkstyle version where as before sevntu overrode the checkstyle version itself. Right?

Correct. When checkstyle dependency was of `compile` scope within sevntu-checks, that checkstyle was used inside `sevntu-checkstyle-maven-plugin` (because of the maven dependency mediation). With checkstyle dependency of `provided` scope, checkstyle from `maven-checkstyle-plugin` will be used instead.

> If these don't include it, where does it get it's version of checkstyle from?

For `sevntu-checkstyle-maven-plugin` see the answer above. For `sevntu-checkstyle-idea-extension` and `eclipsecs-sevntu-plugin` checkstyle was taken from the main plugins.

> Is it like sonar and expects main checkstyle plugin to also be installed already?

Yes, all extensions can not be used without their corresponding main plugins.

Reply:
Thanks. I am not familar with how they worked so thats why I was asking.
I was also looking for how the plugins would be affected by this removal that may change functionality.

So `maven-checkstyle-plugin` will be affected if they don't override checkstyle version.
Our plugins will remain the same because they can't work without checkstyle version.